src/hb-ot-shape-normalize.cc

```diff
@@ -237,8 +237,16 @@ handle_variation_selector_cluster (const hb_ot_shape_normalize_context_t *c,
 	/* Just pass on the two characters separately, let GSUB do its magic. */
 	set_glyph (buffer->cur(), font);
 	(void) buffer->next_glyph ();
-	set_glyph (buffer->cur(), font);
-	(void) buffer->next_glyph ();
+	if (buffer->not_found_variation_selector != HB_CODEPOINT_INVALID)
+	{
+	  _hb_glyph_info_clear_default_ignorable (&buffer->cur());
```

Comment:
Correct. Otherwise it would get replaced by a space glyph or removed completely.

Reply:
Though clearning default-ignorable might have shaping implications as it becomes non-skippable. Maybe we have another flag that I should use. Good catch.

Reply:
One problem with this approach is that it disables GSUB shaping of variation-selectors. Upon further thought, I think that's fine and clearing default-ignorable is fine too, as we're declaring the cluster broken anyway, so any chances to the GSUB skipping is a non-issue.

Reply:
Some fonts implement variation selectors using GSUB ligatures, would they be afftcted by this change?

Reply:
You are indeed right.

My memory is that only the Mongolian-variation-selectors are commonly handled in GSUB. Those are not affected by this code. I think this change is saying: "cmap format14 is the way to declare support for variation selectors". Do you think that's a problem?

Reply:
That would be an unfortunate limitation. I’m pretty sure I have seen fonts that use GSUB for variation selectors other than Mongolian. The tooling around cmap format 14 can be clumsy and some font designers might even know about it, so GSUB might seem like an obvious solution.

Reply:
Let me see if I can modify the implementation. I think we can keep the API for now though, since it will be a while before it will be used anyway. If I was to implement it properly, I'll need an extra bit to remember a glyph is a variation-selector. That's annoying.

Reply:
Alternatively, I can revert this.

Reply:
There's no bits available. However, the 5 bits that store General_Category encode 30 values currently, so there's two values available to be (ab)used. It would be ugly but can be worked out. Let me try...

Reply:
That's really ugly. I don't like it.

The other option is, if the font maps the variation-selector, assume it knows what it's doing and not replace it. That's not very satisfactory but might be an okay compromise. @drott